pkg/cli/mirror/create.go

```diff
@@ -166,14 +193,72 @@ func (o *MirrorOptions) createOlmArtifactsForOCI(ctx context.Context, cfg v1alph
 			return err
 		}
 
+		// setup where the FBC content will be written to
 		catalogContentsDir := filepath.Join(artifactsFolderName, ctlg.Ref.Name)
+		// obtain the path to where the OCI image reference resides
+		layoutPath := layout.Path(v1alpha2.TrimProtocol(ctlg.OCIFBCPath))
+
+		// get its index.json and obtain its manifest
+		rootIndex, err := layoutPath.ImageIndex()
+		if err != nil {
+			return err
+		}
+		rootIndexManifest, err := rootIndex.IndexManifest()
+		if err != nil {
+			return err
+		}
+
+		// attempt to find the first image reference in the layout...
+		// for a manifest list only search one level deep.
+		var img v1.Image
+	loop:
```

Comment:
Labels are optional for break statements, is there any specific reasons of using label (loop) here? If there is not specific reason, it is possible simplying the code removing these labels.


```suggestion
	for _, descriptor := range rootIndexManifest.Manifests {

			if descriptor.MediaType.IsIndex() {
				// follow the descriptor using its digest to get the referenced index and its manifest
				childIndex, err := rootIndex.ImageIndex(descriptor.Digest)
				if err != nil {
					return err
				}
				childIndexManifest, err := childIndex.IndexManifest()
				if err != nil {
					return err
				}

				// at this point, find the first image and store it for later if possible
				for _, childDescriptor := range childIndexManifest.Manifests {
					if childDescriptor.MediaType.IsImage() {
						img, err = childIndex.Image(childDescriptor.Digest)
						if err != nil {
							return err
						}
						// no further processing necessary
						break
					}
				}

			} else if descriptor.MediaType.IsImage() {
				// this is a direct reference to an image, so just store it for later
				img, err = rootIndex.Image(descriptor.Digest)
				if err != nil {
					return err
				}
				// no further processing necessary
				break
			}
		}
```

Reply:
Simply removing the loop label is insufficient for what this code needs to do. We need to stop processing as soon as we find an appropriate image. If I don't use a label I need to put other control structures in place to break out of the inner loop and then break out of the outer loop, or refactor this into its own function. So the label does serve a purpose. Let me know if you'd prefer that I refactor this.

Reply:
Labels are optional for break statements, is there any specific reasons of using label (loop) here? If there is not specific reason, it is possible simplying the code removing these labels.


```suggestion
	for _, descriptor := range rootIndexManifest.Manifests {

			if descriptor.MediaType.IsIndex() {
				// follow the descriptor using its digest to get the referenced index and its manifest
				childIndex, err := rootIndex.ImageIndex(descriptor.Digest)
				if err != nil {
					return err
				}
				childIndexManifest, err := childIndex.IndexManifest()
				if err != nil {
					return err
				}

				// at this point, find the first image and store it for later if possible
				for _, childDescriptor := range childIndexManifest.Manifests {
					if childDescriptor.MediaType.IsImage() {
						img, err = childIndex.Image(childDescriptor.Digest)
						if err != nil {
							return err
						}
						// no further processing necessary
						break
					}
				}

			} else if descriptor.MediaType.IsImage() {
				// this is a direct reference to an image, so just store it for later
				img, err = rootIndex.Image(descriptor.Digest)
				if err != nil {
					return err
				}
				// no further processing necessary
				break
			}
		}
```

Reply:
Hi @jchunkins,

I did not see about the inner loop, now it makes senses for me to keep the label, just ignore my suggestion.